deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java

```diff
@@ -461,7 +480,7 @@ void process(String envId, EnvironBean env, PingReportBean report, AgentBean age
         AgentBean updateBean = null;
         if (report != null) {
             updateBean = genUpdateBeanByReport(report, agent);
-            if (!StringUtils.isEmpty(report.getEnvId())) {
+            if (!StringUtils.isEmpty(report.getEnvId()) && updateBean != null) {
```

Comment:
we are changing the logic here a lot, wondering if we can minimize the logic change and only handle not to update the DB part.  The unit test is failing so worth to try to isolate the change.

Reply:
yes, agree. I will upload new diff.